### ISSUE TITLE
[bug]Webhooks don't work

### ISSUE THREAD

**Issue by DenisDoroshchuk** [vikaspotluri123/discourse-on-ghost#219]
Hello,

So I have Discourse, I also have Ghost CMS. I'm trying to connect them using DoG. I have done everything step by step from the [instructions](https://github.vikaspotluri.me/discourse-on-ghost/). 
SSO works fine, I can access Discourse through the Ghost site. Thanks!
But webhooks I can't get it to work.

My `.env` file:
<img width="1129" alt="Screenshot 2024-10-29 at 13 25 29" src="https://github.com/user-attachments/assets/62ef024f-48b6-4979-8663-9440f56d8cad">

My webhooks in Ghost:
<img width="972" alt="Screenshot 2024-10-29 at 13 29 08" src="https://github.com/user-attachments/assets/4f9d84c9-35d3-4c2d-929f-67b8ca028f15">

I change the username in Ghost but nothing happens in Discourse, the name doesn't change but the hook is sent. This is what I see in Ghost logs (I changed my domain to mysite.com):
`{"name":"Log","hostname":"ghost-staging","pid":263833,"level":30,"version":"5.88.2","msg":"Triggering webhook for \"member.edited\" with url \"https://mysite.com/ghost/api/external_discourse_on_ghost/hook/a5dce7f74fca6254d6115419\"","time":"2024-10-29T12:18:55.622Z","v":0}
{"name":"Log","hostname":"ghost-staging","pid":263833,"level":40,"version":"5.88.2","msg":"Request to https://mysite.com/ghost/api/external_discourse_on_ghost/hook/a5dce7f74fca6254d6115419 failed because of: ERR_NON_2XX_3XX_RESPONSE.","time":"2024-10-29T12:18:55.653Z","v":0}
`

Can you help me figure this out? Again, SSO works fine, i.e. the path `/ghost/api/external_discourse_on_ghost/` is configured in nginx.

Thank you,
Denis D.

**Comment by DenisDoroshchuk** [comment]
Hey @vikaspotluri123 
Thanks for the quick reply. Yes indeed, changing the tier does work. 
I tried logging out of Discourse and logging in again via SSO, but the username did not change.

**Comment by vikaspotluri123** [comment]
Ghost doesn't have the concept of a username for members, right? As part of SSO, the email, avatar, and potentially name are synced.

**Comment by DenisDoroshchuk** [comment]
@vikaspotluri123 Sorry for the confusion, I may have worded it oddly. I would like to be able to change the Name field in Discourse after changing the Name field in Ghost.
<img width="1714" alt="Screenshot 2024-10-30 at 08 21 00" src="https://github.com/user-attachments/assets/7034653f-e658-446d-a7fc-0c9cc5853724">

**Comment by vikaspotluri123** [comment]
Thanks! I'll need to debug this a bit so I won't be able to get to it until this weekend or next weekend

**Comment by DenisDoroshchuk** [comment]
Thank you @vikaspotluri123 ! I will wait answer from you.

**Comment by vikaspotluri123** [comment]
I'm able to reproduce this behavior, thanks for reporting! The member data sent by DoG is correct, so I need to debug why Discourse is ignoring it (it's probably an issue with DoG).

**Comment by DenisDoroshchuk** [comment]
Thank you @vikaspotluri123 
I checked it out and it does work fine. But only if the user logs out and logs in again via SSO.
This setting:
<img width="1511" alt="Screenshot 2024-11-07 at 10 13 03" src="https://github.com/user-attachments/assets/94bdff6c-8f36-4caa-83e3-a5cf72a77c54">

**Comment by vikaspotluri123** [comment]
> I checked it out and it does work fine. But only if the user logs out and logs in again via SSO.

Yes, that's a [current limitation](https://github.com/vikaspotluri123/discourse-on-ghost/issues/219#issuecomment-2444534792):

> But the other thing is that the webhooks only handle subscription changes today; for user changes to sync I think you they will need to log out/in to Discourse

**Comment by vikaspotluri123** [comment]
The docs have been updated to highlight this behavior. Keeping this issue open to track the enhancement.
